src/main.ts

```diff
@@ -35,17 +40,63 @@ export class Minifier {
     }
   }
 
+  setTypeChecker(typeChecker: ts.TypeChecker) {
+    this.typeChecker = typeChecker;
+  }
+
   // Recursively visits every child node, emitting text of the sourcefile that is not a part of
   // a child node.
   visit(node: ts.Node) {
     switch (node.kind) {
       case ts.SyntaxKind.PropertyAccessExpression: {
+        let pae = <ts.PropertyAccessExpression>node;
+        let exprSymbol = this.getExpressionSymbol(pae);
         let output = '';
-        let propAccessExp = <ts.PropertyAccessExpression>node;
-        output += this.visit(propAccessExp.expression);
-        output += '.';
-        // Adds '$mangled' to an identifier to ensure that the identifier is being altered.
-        output += propAccessExp.name.text + '$mangled';
+        let children = pae.getChildren();
+
+        for (var child of children) {
```

Comment:
Do you need the loop here? You know exactly what children a PAE has, don't you? What about just appending the text up to `pae.identifer.getStart()`, then doing the renaming, and be done?